Library/include/CSP/Systems/Organizations/Organization.h

```diff
@@ -0,0 +1,160 @@
+﻿/*
+* Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "CSP/CSPCommon.h"
+#include "CSP/Common/Array.h"
+#include "CSP/Common/String.h"
+#include "CSP/Systems/WebService.h"
+
+
+namespace csp::services
+{
+
+CSP_START_IGNORE
+template <typename T, typename U, typename V, typename W> class ApiResponseHandler;
+CSP_END_IGNORE
+
+} // namespace csp::services
+
+
+namespace csp::systems
+{
+// enum class TierNames;
+
+// @ingroup Organization System
+/// @brief Data representation of User Role in Organization.
+enum class EOrganizationRole
+{
+	Member,
+	Administrator,
+	Owner
+};
+
+/// @ingroup Organization System
+/// @brief Data representation of a User's Organization invite.
+class CSP_API InviteOrganizationRoleInfo
+{
+public:
+	InviteOrganizationRoleInfo() = default;
+
+	csp::common::String UserEmail;
+	csp::common::Array<EOrganizationRole> OrganizationRoles;
+};
+
+/// @ingroup Organization System
+/// @brief Data representation of a collection of User Organization invites.
+class CSP_API InviteOrganizationRoleCollection
+{
+public:
+	InviteOrganizationRoleCollection() = default;
+
+	csp::common::String EmailLinkUrl;
+	csp::common::String SignupUrl;
+	csp::common::Array<InviteOrganizationRoleInfo> InvitedUserRoles;
+};
+
+/// @ingroup Organization System
+/// @brief Data representation of a User's role within an Organization.
+class CSP_API OrganizationRoleInfo
+{
+public:
+	OrganizationRoleInfo() = default;
+
+	csp::common::String UserId;
+	csp::common::Array<EOrganizationRole> OrganizationRoles;
+};
+
+/// @ingroup Organization System
+/// @brief Data representation of an Organization.
+class CSP_API Organization
+{
+public:
+	Organization() = default;
+	Organization(const Organization& Other) = default;
+
+	csp::common::String Id;
+	csp::common::String OwnerId;
+	csp::common::String CreatedAt;
+	csp::common::String CreatedBy;
+	csp::common::String Name;
+	csp::common::String Description;
+	csp::common::Array<OrganizationRoleInfo> Members;
+	int32_t SpaceCount;
+};
+
+/// @ingroup Organization System
+/// @brief Data class used to contain Organization object.
+class CSP_API OrganizationResult : public csp::systems::ResultBase
```

Comment:
Which OrganizationSystem API returns this result ? Based off this diagram here https://miro.com/app/board/uXjVNz7XMkQ=/?moveToWidget=3458764577696164303&cot=14
This would be needed as part of showing the user their organization name requirement.
Or potentially there are other means of getting the organization name?

Reply:
The diagram is out of date and will be updated. They will be able to call GetOrganization using the cached Id.

Reply:
Which API is GetOrganization as i am not spotting it in the system public endpoints ?

Reply:
It hasn't been done yet. This PR relates to tickets OF-1232 and OF-1236 - see here: https://magnopus.atlassian.net/jira/software/c/projects/OF/boards/258?selectedIssue=OF-1207

Tickets still to be worked on are: OF-1233, OF-1237 and OF-1238.
The GetOrganization() endpoint will be part of OF-1237.

Reply:
Right, that makes sense, thank you for the clarification 🙂